MF-268 - Replace JWT with plain string tokens in clients service

anovakovic01 opened the pull request:
This pull request resolves #268.

**Summary:**
- Replace JWT identity provider with postgres identity provider
- Update existing tests
- Add tests for postgres identity provider

Commit:
Replace JWT identity provider implementation with postgres one

Signed-off-by: Aleksandar Novakovic <aleksandar.novakovic@mainflux.com>

Commit:
Add identity provider tests

Add identity provider tests and fix existing postgres tests.

Signed-off-by: Aleksandar Novakovic <aleksandar.novakovic@mainflux.com>

Commit:
Update tests in accordance with identity provider changes

Signed-off-by: Aleksandar Novakovic <aleksandar.novakovic@mainflux.com>

mijicd commented on clients/hasher.go:
```diff
@@ -0,0 +1,12 @@
+
+// Hasher specifies an API for generating hashes of an arbitrary textual
+// content.
+type Hasher interface {
```
Why do you need this?

mijicd commented on clients/postgres/idp.go:
```diff
@@ -0,0 +1,37 @@
+
+var _ clients.IdentityProvider = (*identityProvider)(nil)
+
+type identityProvider struct {
```
This is over-engineering. `Identity` check can be performed as part of the queries in postgres repository, while Key generation can be kept inside the repository (as before).

Commit:
Move identity method from identity provider to client repo

Remove identity provider as it is not needed anymore. Update tests
in accordance with these changes.

Signed-off-by: Aleksandar Novakovic <aleksandar.novakovic@mainflux.com>
